ArduCopter/events.cpp

```diff
@@ -85,20 +85,46 @@ void Copter::handle_battery_failsafe(const char *type_str, const int8_t action)
     }
 }
 
+bool Copter::should_run_failsafe_checks()
+{
+    if (failsafe.gcs()) {
+        // we must run the failsafe checks if we are in failsafe -
+        // otherwise we will never leave failsafe
+        return true;
+    }
+
+    if (g.failsafe_gcs == FS_GCS_DISABLED) {
+        // simply disabled
+        return false;
+    }
+
+    if (failsafe.last_heartbeat_ms == 0) {
+        // GCS has never connected
+        return false;
+    }
+
+    if (RC_Channels::has_active_overrides()) {
+        // GCS is currently telling us what to do!
+        return true;
+    }
+    if (control_mode == GUIDED || control_mode == GUIDED_NOGPS) {
+        // GCS is currently telling us what to do!
+        return true;
+    }
+
+    return false;
+}
+
 // failsafe_gcs_check - check for ground station failsafe
 void Copter::failsafe_gcs_check()
 {
-    uint32_t last_gcs_update_ms;
-
-    // return immediately if gcs failsafe is disabled, gcs has never been connected or we are not overriding rc controls from the gcs and we are not in guided mode
-    // this also checks to see if we have a GCS failsafe active, if we do, then must continue to process the logic for recovery from this state.
-    if ((!failsafe.gcs)&&(g.failsafe_gcs == FS_GCS_DISABLED || failsafe.last_heartbeat_ms == 0 || (!RC_Channels::has_active_overrides() && control_mode != GUIDED))) {
+    if (!should_run_failsafe_checks()) {
         return;
     }
 
     // calc time since last gcs update
     // note: this only looks at the heartbeat from the device id set by g.sysid_my_gcs
-    last_gcs_update_ms = millis() - failsafe.last_heartbeat_ms;
+    uint32_t last_gcs_update_ms = millis() - failsafe.last_heartbeat_ms;
```

Comment:
const. But, why not just remove it?

Reply:
const added.  Could remove the temp, but it fits with the comments above it and I don't feel the need to rewrite the comment to be "check time since last gcs update" etc etc....